workflow/Snakefile

```diff
@@ -16,8 +17,18 @@ container: "docker://condaforge/mambaforge:4.10.1-0"
 SCENARIOS = pd.read_csv(config["scenarios"]).set_index('name')
 GROUPS = pd.read_csv(config['parameters'])['group'].unique()
 
-# Calculates number of model runs for the Method of Morris
-MODELRUNS = range((len(GROUPS) + 1) * config['replicates'])
+PARAMETERS = pd.read_csv(config['parameters'])['indexes'].unique()
+
+METHOD = config['method']
+
+# Calculates number of model runs for the method of choice
+if METHOD == 'Morris':
+    MODELRUNS = range((len(GROUPS) + 1) * config['replicates'])
+elif METHOD == 'LHS':
+    MODELRUNS = range(config['replicates'])
+elif METHOD == 'Sobol':
+    MODELRUNS = range((config['replicates']**2) * (len(PARAMETERS) + 2))
```

Comment:
`len(PARAMETERS)` is always 1, you need:

`PARAMETERS = pd.read_csv(config['parameters'])['name'].to_numpy()`

and 

`MODELRUNS = range((2 ** REPLICATES) * (PARAMETERS.shape[0] + 2))`

Reply:
addressed partially, I did not see the reason for using PARAMETERS.shape[0] as the PARAMETERS have just one dimension. I also kept the PARAMETERS definition linked to the 'indexes', is there any specific reason why I should switch to the 'name'?

Reply:
I guess the fastest way would be `PARAMETERS = len(pd.read_csv(config['parameters']))`

Reply:
One other issue is that this ignores grouping of parameters...